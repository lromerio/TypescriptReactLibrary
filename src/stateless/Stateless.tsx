import * as React from "react";
import {style} from "typestyle/lib";
import {Base} from "../styles";
import "../styles.css";

export interface IStatelessProps {
    id?: string;
    content: string;
}

const Stateless: React.StatelessComponent<IStatelessProps> = (props: IStatelessProps) => {
    return <span id={props.id} className={styles}>{props.content}</span>;
};

Stateless.defaultProps = {
    id: "my_id",
};

/**
 * Styles (CSS like) of the component.
 */
const styles = style(Base.text, {
    fontSize: "50px",
    padding: "20px 0px 20px 20px",
});

export default Stateless;
