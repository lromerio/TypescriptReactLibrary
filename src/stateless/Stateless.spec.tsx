import "raf/polyfill";
import "jsdom-global/register"

import * as React from "react"
import {configure, mount, shallow} from "enzyme"
import * as Adapter from "enzyme-adapter-react-16"
import toJson from "enzyme-to-json"

import Stateless from "."


describe("<Stateless />", () => {

    const text = "Test"

    const comp = <Stateless content={text} />

    configure({ adapter: new Adapter() })

    it("should render", () => {
        const wrapper = mount(comp)
        expect(toJson(wrapper)).toMatchSnapshot()
        wrapper.unmount()
    })

    it("should update with props", () => {
        const newText = "New Text"
        
        const wrapper = shallow(comp)

        expect(wrapper.find("span").text()).toBe(text)
        wrapper.setProps({content: newText })
        expect(wrapper.find("span").text()).toBe(newText)

        wrapper.unmount()
    })
})