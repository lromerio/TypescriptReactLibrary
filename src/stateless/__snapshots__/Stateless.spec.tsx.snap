// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stateless /> should render 1`] = `
<Stateless
  content="Test"
  id="my_id"
>
  <span
    className="f1wexwm6"
    id="my_id"
  >
    Test
  </span>
</Stateless>
`;
